src/main/scala/app/interpreter/MutableMapEventStoreInterpreter.scala

```diff
@@ -39,22 +41,22 @@ class MutableMapEventStoreInterpreter(implicit ec: ExecutionContext) extends Eve
             event.systemName.name == systemName.name
         }
         val orderedEvents: List[Event] = eventsForEntity.toList.map(_._2).sortBy(_.suppliedTimestamp)
-        onResult(orderedEvents)
+        orderedEvents
 
-      case SaveSnapshot(snapshot, next) =>
+      case SaveSnapshot(snapshot) =>
         mutableSnapshotMap += (snapshot.id.id -> snapshot)
-        next
+        ()
```

Comment:
Don't like this, but without the () to force the Future to be of type Unit, resulted in a compilation error.  Otherwise could extract to another function with a return type of Unit.